corehq/util/workbook_reading/tests/files/csv/mixed.csv

```diff
@@ -0,0 +1,2 @@
+here,are,four,things
+but,this,row,has,five
```

Comment:
I assume trailing empty cells shouldn't be a problem, right?  Like
```
this,row,has,five,
```

Reply:
I think that example would count as five cells, the last one a blank string, and would then error if all other rows didn't have five cells. But I do think it's fair to call that an error - do you disagree?

This is the code that builds `rows`: https://github.com/dimagi/commcare-hq/blob/1b97f368a64a16fde84bbd49eabb15540d3b46be/corehq/util/workbook_reading/adapters/csv.py#L40-L41

Reply:
No, that sounds correct.  The incorrect scenario I was imagining would be if it counted that example as only 4 cells.  I've seen spreadsheet errors in the past where it behaves weirdly around blank cells or empty rows, since with xls and xlsx for example it's hard to tell how many columns are "real" (especially if you've applied a style or rule to a whole row).  I just tried it out and it looks like `csv.reader` behaves like I'd expect:
```python
In [1]: import csv, io

In [2]: f = io.StringIO(
   ...: """1,2,3,4,,
   ...: 1,2,3,4,
   ...: 1,2,3,4
   ...: """)

In [3]: list(csv.reader(f))
Out[3]: [['1', '2', '3', '4', '', ''], ['1', '2', '3', '4', ''], ['1', '2', '3', '4']]
```
That said, I wouldn't be surprised if some spreadsheet program wrote to csv with a heterogeneous number of commas if there's trailing blank data in there.  Fine to cross that bridge when we come to it though.

Reply:
That all sounds good 👍